File: Day203/rotatedSearchII.go
package  main

import (
	"fmt"
)

func search(nums []int, target int) bool {
    low:=0
    high:=len(nums)-1
    for low<= high{
        mid:= high - ((high-low)>>1)
        if(nums[mid]==target){
            return true
        }
        if(nums[mid]==nums[low] && nums[high]==nums[mid]){
            low++
            high--
        }else if(nums[low]<=nums[mid]){
            if(target>=nums[low] && target<nums[mid]){
                high = mid -1

            }else{
                low = mid+1
            }
        }else{
            if(target<=nums[high] && target>nums[mid]){
                low = mid +1
            }else{
                high = mid-1
            }
        }
        
    }
    return false
}
func main() {

	var arr = []int{1,0,1,1,1,1,1}
	fmt.Println(len(arr))
	fmt.Println(search(arr, 0))
}

File: Day203/rotatedSearchII.java
package Day203;



class Solution {
    public boolean search(int[] nums, int target) {
       int low = 0, high = nums.length - 1;
        while(low <= high){
            int mid = high - ((high - low) >> 1);
            if(nums[mid] == target)
                return true;
            if(nums[mid] == nums[low] && nums[mid] == nums[high]){
                low++;
                high--;
            }
            else if(nums[low] <= nums[mid]){
                if(target >= nums[low] && target < nums[mid])
                    high = mid - 1;
                else
                    low = mid + 1;
            }
            else{
                if(target <= nums[high] && target > nums[mid])
                    low = mid + 1;
                else
                    high = mid - 1;
            }
        }
        return false;
    }

    
}

public class rotatedSearchII{
    public static void main(String[] args){
        int arr[] = {2,5,6,0,0,1,2};
        Solution sol = new Solution();
        System.out.println(sol.search(arr, 0));
    }
}